clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -1245,6 +1248,9 @@ public void handle(OffsetFetchResponse response, RequestFuture<Map<TopicPartitio
 
             if (unauthorizedTopics != null) {
                 future.raise(new TopicAuthorizationException(unauthorizedTopics));
+            } else if (!pendingTxnOffsetTopicPartitions.isEmpty()) {
+                // just retry
+                future.raise(new PendingTransactionException(pendingTxnOffsetTopicPartitions));
```

Comment:
Since this exception can still be thrown out of the public API (e.g. in `consumer.committed` if we exhaust the timeout before retrying succeed) we should also update the javadoc accordingly; and also we should update the KIP for this API change as well.

Reply:
EDIT: actually the one to throw is still `TimeoutException` so we do not need to update javadoc.